scripts/core/editor3/components/Editor3.jsx

```diff
@@ -20,146 +21,137 @@ import Toolbar from './toolbar';
  * @description Editor3 is a draft.js based editor that support customizable
  *  formatting (shortly more features will come ...)
  */
-export class Editor3 extends React.Component {
-    constructor(props) {
-        super(props);
 
-        const initialContentState = stateFromHTML(props.value);
+export const getInitialState = (spellcheckerService, value, onChangeValue, readOnly, showToolbar, editorFormat) => {
+    const initialContentState = stateFromHTML(value);
 
-        this.state = {
-            editorState: EditorState.createWithContent(
+    return {
+        editorState: EditorState.createWithContent(
```

Comment:
This is a state object, so calling properties with the `State` suffix is stuttering imo. `editor` should be enough, and below, `spellchecker` should also be enough.